sql/sql_string.h

```diff
@@ -701,6 +701,9 @@ class Binary_string: public Sql_alloc
   bool realloc_raw(size_t arg_length);
   bool realloc(size_t arg_length)
   {
+    // This increment is redundant because we allocate the size incremented 
+    // inside realloc_raw. Removing this +1, however, introduces a lot of test failures
+    // so we leavet it for now. It does no harm to allocate one extra byte.
```

Comment:
> Removing this +1, however, introduces a lot of test failures

What kind of test failures?

If this `+1` appears logically unnecessary, but removing it causes failures, that suggests to me that there must be an off-by-`-1` error somewhere in `realloc_raw`, which the `+1` here is mitigating by wrapping it.

Reply:
```
worker[1] Using MTR_BUILD_THREAD 300, with reserved ports 16000..16019
main.pool_of_threads                     [ fail ]
        Test ended at 2022-05-22 14:00:32

CURRENT_TEST: main.pool_of_threads
mysqltest: In included file "./include/common-tests.inc":
included from /home/norakagi/projects/mariadb-dev/server/MDEV-28315/mysql-test/main/pool_of_threads.test at line 12:
At line 13: query 'drop table if exists t1,t2,t3,t4' failed: <Unknown> (5): Out of memory (Needed 24 bytes)

The result from queries just before the failure was:
SET optimizer_switch='outer_join_with_cache=off';
drop table if exists t1,t2,t3,t4;

 - saving '/home/norakagi/projects/mariadb-dev/server/MDEV-28315/bld/mysql-test/var/log/main.pool_of_threads/' to '/home/norakagi/projects/mariadb-dev/server/MDEV-28315/bld/mysql-test/var/log/main.pool_of_threads/'
--------------------------------------------------------------------------
The servers were restarted 0 times
Spent 0.000 of 2 seconds executing testcases

Failure: Failed 1/1 tests, 0.00% were successful.
```

One failure I found is as above. You are correct but that I think should be a separate issue.

Reply:
Agreed. It's a somewhat separate issue.